tests/test_controller.py

```diff
@@ -3109,6 +3061,96 @@ def test_permalink_returns_fulfillment_links_for_authenticated_patrons_with_loan
         eq_(expect, response.data)
         eq_(OPDSFeed.ENTRY_TYPE, response.headers['Content-Type'])
 
+    def test_permalink_returns_fulfillment_links_for_authenticated_patrons_with_fulfillment(self):
+        auth = dict(Authorization=self.valid_auth)
+
+        with self.request_context_with_library("/", headers=auth):
+            content_link = 'https://content'
+
+            # We have two patrons.
+            patron_1 = self.controller.authenticated_patron(self.valid_credentials)
+            patron_2 = self._patron()
+
+            # But the request was initiated by the first patron.
+            flask.request.patron = patron_1
+
+            identifier_type = Identifier.GUTENBERG_ID
+            identifier = '1234567890'
+            edition, _ = self._edition(
+                title='Test Book',
+                identifier_type=identifier_type,
+                identifier_id=identifier,
+                with_license_pool=True
+            )
+            work = self._work(
+                'Test Book',
+                presentation_edition=edition,
+                with_license_pool=True
+            )
+            pool = work.license_pools[0]
+            [delivery_mechanism] = pool.delivery_mechanisms
+
+            loan_info = LoanInfo(
+                pool.collection, pool.data_source.name,
+                pool.identifier.type,
+                pool.identifier.identifier,
+                datetime.datetime.utcnow(),
+                datetime.datetime.utcnow() + datetime.timedelta(seconds=3600),
+            )
+            self.manager.d_circulation.queue_checkout(
+                pool,
+                loan_info
+            )
+
+            fulfillment = FulfillmentInfo(
```

Comment:
It should no longer be necessary to create a FulfillmentInfo or a Resource as part of test setup, since it's no longer considered when generating the OPDS.

Reply:
The following code requires a `FulfillmentInfo` object:
```python
self.manager.loans.fulfill(
    pool.id, delivery_mechanism.delivery_mechanism.id,
)
```
It might make sense to remove the whole test case but I would like to still keep it to make sure that the code correctly processes works with fulfillments (by just ignoring them).